runtime/onert/backend/train/KernelGenerator.cc

```diff
@@ -418,16 +418,21 @@ void KernelGenerator::visit(const ir::train::operation::Loss &node)
     {
       const auto y_pred_op_code = node.y_pred_op_code();
       bool is_normalization_required = (y_pred_op_code != ir::OpCode::Softmax);
+      const auto cce_params = std::get_if<ir::train::CategoricalCrossentropyParam>(&loss_param);
+      if (!cce_params)
+      {
+        throw std::runtime_error("LossLayer: Expected loss_param to be "
+                                 "CategoricalCrossentropyParam but found a different type.");
+      }
```

Comment:
Using `std::get_if` to check the `variant`’s type adds a very small runtime overhead due to the type check. But the safety and clarity improvements are generally worth the minimal performance impact.